pkg/services/ngalert/state/template.go

```diff
@@ -87,3 +86,20 @@ func newTemplateCaptureValues(values map[string]eval.NumberValueCapture) map[str
 	}
 	return m
 }
+
+func buildExploreLink(showGraph, showTable bool) func(string) string {
```

Comment:
I'm not sure how I feel about this functions' signature. I think we should try to avoid boolean flags. Nobody without the context of the function can tell what will happen in line 61 & 62. If you want to reuse it at all costs, you could use the option pattern (`buildExploreLink(withGraph())`, `buildExploreLink(withTable())`). Otherwise, you can just make it two functions. 

I don't have any hard feelings against this, I personally just try to avoid giving people a hard time to understand what's going on (here with the context right now it's easy, but revisiting after half a year is another story).

WDYT?

Reply:
Thought the same thing when writing this function. I was expecting some feedback on this haha.
@yuri-tceretian @gerobinson any thoughts?

Reply:
I agree that this is not perfectly readable. I am ok with the existing code considering the fact that it is not a public API and the scope is limited.